### PR TITLE
fix: fixes inability to rename other widgets, jsactions, pages

### LINKED ISSUE #9843: [Bug]: App crash on edit name on Entity Explorer
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

[LOOM DEMO](https://www.loom.com/share/1760f15e0067408c9c46fb9d42b8d414)

### Steps To Reproduce

Try to edit any name in entity explorer, working fine for datasources.

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by Irongade**
> Pull Request Template
App breaks anytime you try to rename a widget, a page or a jsObject, This PR fixes that

## Description
The bug was as a result of an empty array being passed to one of the if else statements which made the wrong conditional block to run.

Fixes #9843

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/fix-widget-naming-bug 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/selectors/entitiesSelector.ts | 53.52 **(0)** | 13.13 **(0.63)** | 33.07 **(0)** | 50.81 **(0)**</details>

### CODE REVIEW

File: app/client/src/selectors/entitiesSelector.ts [hunk 1]
```diff
@@ -507,10 +507,12 @@ export const getExistingActionNames = createSelector(
...
     // if the current action being edited is on the same page, filter the actions on the page and return their names.
+    // or if the there is no current action being edited (this happens when a widget, or any other entity is being edited), return the actions on the page.
     if (
-      editingAction &&
-      editingAction.length > 1 &&
-      editingAction[0].config.pageId === currentPageId
+      (editingAction &&
+        editingAction.length > 1 &&
+        editingAction[0].config.pageId === currentPageId) ||
+      (editingAction && editingAction.length < 1)
```
Comment (nidhi-nair) [inline app/client/src/selectors/entitiesSelector.ts:515]: What's supposed to happen when the length is 1?

Comment (Irongade) [reply app/client/src/selectors/entitiesSelector.ts:515]: should be 0 not 1, this has been resolved.

**Review by AmanAgarwal041 — APPROVED** [review]

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**